From 07a2c381e79f96825d69dffd0a37034228ebd65d Mon Sep 17 00:00:00 2001
From: Misael Lopez Cruz <misael.lopez@ti.com>
Date: Tue, 14 Dec 2010 19:09:17 -0600
Subject: ASoC: sdp4430: Set cpu_dai format for Multimedia FM Digital port

Set cpu_dai format (I2S) for Multimedia FM Digital port (legacy).
Multimedia FM Digital port is a direct connection between CPU and
McBSP2 port, ABE is in bypass mode.

Change-Id: Ief82cbb7b9a5aeafed99b51d7db35bc5714477bd
Signed-off-by: Misael Lopez Cruz <misael.lopez@ti.com>
---
 sound/soc/omap/sdp4430.c |   17 +++++++----------
 1 files changed, 7 insertions(+), 10 deletions(-)

Index: kernel/sound/soc/omap/sdp4430.c
===================================================================
--- kernel.orig/sound/soc/omap/sdp4430.c	2011-04-05 21:57:24.000000000 +0900
+++ kernel/sound/soc/omap/sdp4430.c	2011-04-07 00:40:12.000000000 +0900
@@ -170,19 +170,17 @@
 {
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
 	struct snd_soc_dai *cpu_dai = rtd->cpu_dai;
+	unsigned int be_id = rtd->dai_link->be_id;
 	int ret = 0;
-	unsigned int be_id;
-
-        be_id = rtd->dai_link->be_id;
 
-	if (be_id == OMAP_ABE_DAI_MM_FM) {
-		/* Set cpu DAI configuration */
+	if (be_id == OMAP_ABE_DAI_BT_VX) {
+		 ret = snd_soc_dai_set_fmt(cpu_dai,
+                                   SND_SOC_DAIFMT_DSP_B |
+                                   SND_SOC_DAIFMT_NB_IF |
+                                   SND_SOC_DAIFMT_CBM_CFM);
+		}
+	else{
 		ret = snd_soc_dai_set_fmt(cpu_dai,
-				  SND_SOC_DAIFMT_I2S |
-				  SND_SOC_DAIFMT_NB_NF |
-				  SND_SOC_DAIFMT_CBM_CFM);
-	} else if (be_id == OMAP_ABE_DAI_BT_VX) {
-	        ret = snd_soc_dai_set_fmt(cpu_dai,
 		/* 
 		 * BT chip and modem uses I2S protocal during voicecall. 
 		 * So McBSP1 also should use I2S. 
